src/debug/HyprCtl.cpp

```diff
@@ -580,11 +580,13 @@ std::string devicesRequest(eHyprCtlOutputFormat format, std::string request) {
         "variant": "{}",
         "options": "{}",
         "active_keymap": "{}",
+        "capsLock": {},
+        "numLock": {},
         "main": {}
     }},)#",
                 (uintptr_t)k.get(), escapeJSONStrings(k->hlName), escapeJSONStrings(k->currentRules.rules), escapeJSONStrings(k->currentRules.model),
                 escapeJSONStrings(k->currentRules.layout), escapeJSONStrings(k->currentRules.variant), escapeJSONStrings(k->currentRules.options), escapeJSONStrings(KM),
-                (k->active ? "true" : "false"));
+                ((k->getLEDs().value_or(0) & (1 << 1)) == 2 ? "true" : "false"), ((k->getLEDs().value_or(0) & (1 << 0)) ? "true" : "false"), (k->active ? "true" : "false"));
```

Comment:
why LEDs? Just check the modifier mask in the current mod state of the keyboard (`modifiersState`)

LEDs are not reliable